src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      const index = Animal.alive.indexOf(this);
+
+      if (index !== -1) {
+        Animal.alive.splice(index, 1);
+      }
+    }
+  }
```

Comment:
The `checkHealth` method uses nested loops (`indexOf` + `splice`) which can be inefficient. It's better to use `filter` to recreate the `Animal.alive` array without the dead animal. This approach avoids the potential performance hit of searching through the array and then splicing it.